src/librustc_borrowck/diagnostics.rs

```diff
@@ -138,6 +138,30 @@ Book:
 https://doc.rust-lang.org/book/ownership.html
 "##,
 
+E0383: r##"
+This error occurs when an attempt is made to partially reinitialize a
+structure that is currently uninitialized.
+
+For example, this can happen when a transfer of ownership has taken place:
+
+```
+let mut t = Test { a: 1, b: None};
+let mut u = Test { a: 2, b: Some(Box::new(t))}; // `t` is now uninitialized
```

Comment:
Can we just `drop(t)` or move `t` elsewhere here (e.g. `let u = Box::new(t)`)? The recursive type adds to the confusion.